src/LodeRunner.API.Test/IntegrationTests/Controllers/TestRuns.cs

```diff
@@ -0,0 +1,156 @@
+        /// <param name="output">The output.</param>
+        public TestRuns(ApiWebApplicationFactory<Startup> factory, ITestOutputHelper output)
+        {
+            this.factory = factory;
+
+            this.output = output;
+
+            this.jsonOptions = new ()
+            {
+                IgnoreNullValues = true,
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+                DictionaryKeyPolicy = JsonNamingPolicy.CamelCase,
+                PropertyNameCaseInsensitive = true,
+            };
+            this.jsonOptions.Converters.Add(new JsonStringEnumConverter());
+        }
+
+        /// <summary>
+        /// Gets endpoints return success and correct content.
+        /// </summary>
+        /// <param name="url">The URL.</param>
+        /// <param name="expectedValue">The expectedValue.</param>
+        /// <returns><see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Theory]
+        [Trait("Category", "Integration")]
+        [InlineData("/", "text/html; charset=utf-8")]
+        [InlineData("/version", "text/plain")]
+        public async Task CanGetEndpointsReturnSuccessAndCorrectContentType(string url, string expectedValue)
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            var httpResponse = await httpClient.GetAsync(url);
+
+            if (httpResponse.IsSuccessStatusCode)
+            {
+                Assert.Equal(expectedValue, httpResponse.Content.Headers.ContentType.ToString());
+            }
+            else
+            {
+                Assert.True(false, $"Unable to process {nameof(this.CanGetEndpointsReturnSuccessAndCorrectContentType)} request.");
+            }
+        }
+
+        /// <summary>
+        /// Determines whether this instance [can get test runs].
+        /// </summary>
+        /// <returns><see cref="Task"/> representing the asynchronous unit test.</returns>
+        [Fact]
+        [Trait("Category", "Integration")]
+        public async Task CanGetTestRuns()
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            await httpClient.GetTestRuns(TestRunsUri, this.jsonOptions, this.output);
+        }
+
+        /// <summary>
+        /// Determines whether this instance [can post test runs].
+        /// </summary>
+        /// <returns><see cref="Task"/> representing the asynchronous integration test.</returns>
+        [Fact]
+        [Trait("Category", "Integration")]
+        public async Task CanCreateAndGetTestRunById()
+        {
+            using var httpClient = ComponentsFactory.CreateLodeRunnerAPIHttpClient(this.factory);
+
+            var postedTestRun = await httpClient.PostTestRun(TestRunsUri, this.jsonOptions, this.output);
+            var gettedTestRun = await httpClient.GetTestRunById(TestRunsUri + "/" + postedTestRun.Id, this.jsonOptions, this.output);
```

Comment:
We should assert that the response is HTTP 200 on GET with valid content.

Reply:
done